aheckmann/mquery
# add eslint, update coding styles and deps. fix some bugs

A lot of non-breaking improvements

- Add eslint and use the same style configuration with `mongoose`
  except for `es6` and `ecmaVersion` for compability. `mquery` uses ES5.
  Any idea for different styles configuration?
- Bump mocha, sliced, debug, bluebird versions without breaking changes.
- Now test node v10 in travis-ci (for `npm audit` security check, see below)
- Deprecate `nsp check`
  - `nsp` is [shutting down on 2018/09/30](https://blog.npmjs.org/post/175511531085/the-node-security-platform-service-is-shutting) and the test may fail after that date.
  - use `npm audit` now to test on `npm@6+`
    `npm install` automatically run with `npm audit` so no need to add it in test script when using npm@6+ which is the version that travis-ci uses in node v10. See Automattic/mongoose#6514 . If we specify `npm audit` in the test script it will fail on old version of npm.
- Update README.md
  - Place maxTime and maxTimeMS in the same line to show that they are just alias of each other.
  - Use new travis-ci svg badge to support high-resolution image.
  Before: ![Build Status](https://travis-ci.org/aheckmann/mquery.png) After: ![Build Status](https://travis-ci.org/aheckmann/mquery.svg?branch=master)
  - Add npm version and install badges
- Bug Fixes. fix some potential errors with the help of linter.

  eslint: `streamOptions` is defined but not used. missing `.stream(streamOptions)` here
  https://github.com/aheckmann/mquery/blob/c35c64eb1268d388c48757d4002d09ebec8021f9/lib/collection/node.js#L126-L132

  eslint: `field` is undefined. should be `key` here
  https://github.com/aheckmann/mquery/blob/c4de96f3013606c6d2e9d4ea63123f1141ed443e/lib/mquery.js#L1343-L1356

  Actually, #103 is also found by VSCode built-in linter

IMO, add linter is good as it can help to spot errors early without testing.
